packages/bento-design-system/src/Menu/createMenu.tsx

```diff
@@ -119,6 +177,157 @@ export function createMenu(
       </Box>
     );
   };
+
+  function NestedMenu({
+    label,
+    items: _items,
+    isSelected,
+    placement,
+    offset,
+    size,
+    state,
+    triggerRef,
+    overlayRef,
+    closeOnSelect,
+    dividers,
+    maxHeight,
+  }: NestedMenuProps) {
+    const componentRef = useRef(null);
+
+    const { childMenuTriggerRefs, childMenuState, selectedMenuItem, setSelectedMenuItem } =
+      useNestedMenu(_items);
+
+    const items = processMenuItems(
+      _items,
+      state,
+      selectedMenuItem,
+      setSelectedMenuItem,
+      closeOnSelect,
+      dividers,
+      maxHeight,
+      size,
+      placement,
+      offset,
+      childMenuState,
+      childMenuTriggerRefs,
+      overlayRef
+    );
+
+    const { menuProps } = useMenuTrigger({}, state, triggerRef);
+    const { overlayProps: positionProps } = useOverlayPosition({
+      containerPadding: 0,
+      targetRef: triggerRef,
+      overlayRef: componentRef,
+      isOpen: isSelected,
+      placement,
+      offset,
+    });
+
+    useEffect(() => {
+      if (!isSelected) {
+        childMenuState.close();
+        setSelectedMenuItem(undefined);
+      }
+    }, [isSelected, childMenuState, setSelectedMenuItem]);
+
+    const nestedMenuPortal = useMemo(
+      () =>
+        createPortal(
+          <FocusScope restoreFocus>
```

Comment:
Currently focus may not be working perfectly: when opening a sub-menu, `Tab` does not immediately shift focus to the first element of the newly opened sub-menu, but finishes cycling through the elements of the currently selected menu before shifting to the opened sub-menu. I'm not sure which behaviour we'd rather have: intuitively I assume the first, but I couldn't find an easy way to implement it with `FocusScope`.